x509/x509.go

```diff
@@ -1925,6 +1928,20 @@ func parseCertificate(in *certificate) (*Certificate, error) {
 				return nil, err
 			}
 			out.TorServiceDescriptors = descs
+		} else if e.Id.Equal(oidExtCABFOrganizationID) {
+			cabf := CABFOrganizationIDASN{}
+			_, err := asn1.Unmarshal(e.Value, &cabf)
+			if err != nil {
+				return nil, err
+			}
+			out.CABFOrganizationIdentifier = &CABFOrganizationIdentifier{
+				Scheme:    cabf.RegistrationSchemeIdentifier,
+				Country:   cabf.RegistrationCountry,
+				Reference: cabf.RegistrationReference,
+			}
+			if len(cabf.RegistrationStateOrProvince) > 0 {
```

Comment:
Why is this only for State and not also Country, Scheme, and Reference?

Reply:
State is the only one that's optional in the ASN1, and that appears to be implemented as a array of length 0 or 1, rather than a string.

I'm no ASN1 expert, but since the fields aren't tagged and are "implicit", you can't have an optional field without making it an array. Based off of exploring the ASN1 structure on the test cert, that's where I arrived.

```
    CABFOrganizationIdentifier ::= SEQUENCE {
        registrationSchemeIdentifier   PrintableString (SIZE(3)),
        registrationCountry            PrintableString (SIZE(2)),
        registrationStateOrProvince    [0] IMPLICIT PrintableString OPTIONAL (SIZE(0..128)),
        registrationReference          UTF8String
    }
```

Reply:
https://censys.io/certificates/a97f9308f3c297ad22c4197afc99b79b1dfd2dad094ddc8df70ba2634a4f7175/asn.1

Reply:
After some discussion with someone who understands ASN.1, this is actually a tagged optional string, not an array, and I cleaned up the deserialization accordingly.